decidim-bulletin_board-js/src/voter/voter.js

```diff
@@ -14,22 +15,43 @@ export class Voter {
    * @param {Object} params - An object that contains the initialization params.
    *  - {String} id - The voter identifier.
    */
-  constructor({ id, bulletinBoardClient, electionContext, options }) {
+  constructor({ id, electionContext, bulletinBoardClient, options }) {
     this.id = id;
-    this.wrapper = new VoterWrapper({ voterId: id });
-    this.bulletinBoardClient = bulletinBoardClient;
     this.electionContext = electionContext;
+    this.bulletinBoardClient = bulletinBoardClient;
+    this.wrapper = new VoterWrapper({ voterId: id });
+    this.parser = new JWTParser();
     this.options = options || { bulletinBoardWaitTime: WAIT_TIME_MS };
   }
 
+  /**
+   * Performs some operations to setup the voter.
+   *
+   * Retrieves the key ceremony messages needed to cast a vote in the given election.
+   *
+   * @returns {Promise<undefined>}
+   */
+  setup() {
+    return this.bulletinBoardClient
+      .getElectionLogEntries({
+        electionUniqueId: this.electionContext.id,
+      })
+      .then(async (logEntries) => {
+        for (const logEntry of logEntries) {
+          const message = await this.parser.parse(logEntry.signedData);
+          this.wrapper.processMessage(logEntry.messageId, message);
+        }
+      });
+  }
+
   /**
    * Encrypts the data using the wrapper.
    *
    * @param {Object} data - The data to be.
    * @returns {Object} - The data encrypted.
    */
   async encrypt(data) {
-    return this.wrapper.encrypt(data);
+    return await this.wrapper.encrypt(data);
```

Comment:
The `await` is not necessary (because the `async` is not necessary neither). You can just return the method call. Also, update the comment because it seems that it is returning a `Promise<Object>`.